src/compositemark/errorbar.ts

```diff
@@ -1,42 +1,395 @@
+
+  // add warning check in the test in this format.
+  // it('warn if size is data driven and autosize is fit', log.wrap((localLogger) => {
+  //   const spec = compile({
+  //     "data": {"values": [{"a": "A","b": 28}]},
+  //     "mark": "bar",
+  //     "autosize": "fit",
+  //     "encoding": {
+  //       "x": {"field": "a", "type": "ordinal"},
+  //       "y": {"field": "b", "type": "quantitative"}
+  //     }
+  //   }).spec;
+  //   assert.equal(localLogger.warns[0], log.message.CANNOT_FIX_RANGE_STEP_WITH_FIT);
+  //   assert.equal(spec.width, 200);
+  //   assert.equal(spec.height, 200);
+  // }));
+  if ((center === 'median') !== (extent === 'iqr')) {
+    log.warn(`${center} is not usually used with ${extent} for error bar.`);
+  }
 
-  if (!encoding.x2 && !encoding.y2) {
-    throw new Error('Neither x2 or y2 provided');
+  const orient: Orient = errorBarOrient(spec);
+  const {transform, continuousAxisChannelDef, continuousAxis, groupby, encodingWithoutContinuousAxis} = errorBarParams(spec, orient, center, extent);
+
+  const {size: _s, ...encodingWithoutSizeAndContinuousAxis} = encodingWithoutContinuousAxis;
+
+  const {scale, axis} = continuousAxisChannelDef;
+
+  function errorBarSinglePointPartMarksSpec(partName: ErrorBarPart, startingPosition: string = center, endingPosition:string = undefined) {
+    const markProperties: boolean | object = config.errorbar[partName];
+    const defaultMarkDef: Mark | MarkDef = (isBoolean(markProperties)) ? getMarkNameOfErrorBarPart(partName) : {
+      type: getMarkNameOfErrorBarPart(partName),
+      ...markProperties
+    };
+
+    return partLayerMixins<ErrorBarPartsMixins>(
+      markDef, partName, config.errorbar,
+      {
+        mark: defaultMarkDef,
+        encoding: {
+          [continuousAxis]: {
+            field: startingPosition + '_' + continuousAxisChannelDef.field,
+            type: continuousAxisChannelDef.type,
+            ...(scale ? {scale} : {}),
+            ...(axis ? {axis} : {})
+          },
+          ...(isString(endingPosition) ? {
+            [continuousAxis + '2']: {
+              field: endingPosition + '_' + continuousAxisChannelDef.field,
+              type: continuousAxisChannelDef.type
+            }
+          } : {}),
+          ...encodingWithoutSizeAndContinuousAxis
+        }
+      }
+    );
   }
 
   return {
     ...outerSpec,
+    transform,
     layer: [
+      ...errorBarSinglePointPartMarksSpec('bar'),
+      ...errorBarSinglePointPartMarksSpec('line'),
+      ...errorBarSinglePointPartMarksSpec('ticks', 'lower_rule'),
+      ...errorBarSinglePointPartMarksSpec('ticks', 'upper_rule'),
+      ...errorBarSinglePointPartMarksSpec('rule', 'lower_rule', 'upper_rule'),
+      ...errorBarSinglePointPartMarksSpec('point')
+    ]
+  };
+}
+
+function errorBarOrient(spec: GenericUnitSpec<Encoding<Field>, ErrorBar | ErrorBarDef>): Orient {
```

Comment:
This method is exactly the same as boxplot's (except the expected `aggregate`)? 

If so, extract this method to `common.ts` and reuse.